### PR TITLE
Closes #115 Add require logged in user for dashboard path

### LINKED ISSUE #10: Guest is Required to Log In to View Dashboard
As a guest,
When I visit "/dashboard",
I am redirected to the login page

### LINKED ISSUE #115: Add Authorization for Dashboards

### PR DESCRIPTION
**Pull request by danjwinter**
Add callback for dashboard method in users_controller to require logged in user or redirect to login path.

closes #115 and closes #10

### CODE REVIEW

**Comment by jwperry** [comment]
Good catch.

File: test/integration/guest_cannot_view_dashboard_test.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require "test_helper"
+
+class GuestCannotViewDashboardTest < ActionDispatch::IntegrationTest
+  test "guest cannot view dashboard" do
+    visit dashboard_path
+
+    assert_equal login_path, current_path
+  end
+end
```
Comment (ToniRib) [inline test/integration/guest_cannot_view_dashboard_test.rb:9]: There's an easy atom checkbox to ensure a single trailing newline in the whitespace package so you won't get this anymore!

File: app/controllers/user/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class User::UsersController < ApplicationController
   require "states_helper"
   before_action :set_user, only: [:edit, :update]
-  before_action :require_logged_in_user, only: [:show]
+  before_action :require_logged_in_user, only: [:show, :dashboard]
```
Comment (ToniRib) [inline app/controllers/user/users_controller.rb:4]: Looks good, this is an easy fix.

**Comment by ToniRib** [comment]
I'm good with this, let me know when you merge in master and I will pull this in.

**Pull request merged**